### PR TITLE
(User roles) Create useNavigation hook and extract scopes to constants

### PR DESCRIPTION
**Pull request by tumbledwyer**
Create `useNavigation` hook to extract scope logic away from the Navigation component. 
Changed scope list to constants in the commons package so they can be referenced anywhere in the app by name rather than a magic string.

### CODE REVIEW

File: packages/client/src/utils/gateway.ts [hunk 1]
```diff
@@ -89,6 +89,87 @@ export const scopes = [
...
+    'organisation.read-locations:my-office',
+  ORGANISATION_READ_LOCATIONS: 'organisation.read-locations',
+  USER_READ_MY_OFFICE: 'user.read:my-office',
+  USER_READ: 'user.read',
+  USER_CREATE: 'user.create',
+  USER_CREATE_MY_JURISDICTION: 'user.create:my-jurisdiction',
+  USER_UPDATE_MY_OFFICE: 'user.update:my-office',
+  USER_UPDATE: 'user.update'
+} as const
+
```
Comment (Nil20) [inline packages/client/src/utils/gateway.ts:172]: These can't be fetched from commons?

Comment (tumbledwyer) [reply packages/client/src/utils/gateway.ts:172]: Unfortunately not. I spent 2 days trying to get that to work but there's an incompatibility with ES and CommonJS modules in the client project. This seems to be the workaround until we upgrade to Typescript v5.0

File: packages/client/src/components/ScopedComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  children
+}) => {
+  const { hasAnyScope } = usePermissions()
+  const hasRequiredScope = !scopes || hasAnyScope(scopes)
+  const hasDeniedScope = denyScopes && hasAnyScope(denyScopes)
+
+  return hasRequiredScope && !hasDeniedScope ? <>{children}</> : null
+}
+
+export default ScopedComponent
```
[the remark this answers is not in the document]
Comment (tumbledwyer) [reply packages/client/src/components/ScopedComponent.tsx]: I've removed this now and replaced it with the hook as per a discussion thread on slack

File: packages/commons/src/scopes.ts [hunk 3]
```diff
@@ -8,103 +8,114 @@
...
 
   // profile
-  'profile.update-signature',
-  'profile.update-phone-number',
-  'profile.update-name',
-  'profile.update-profile-image',
+  PROFILE_UPDATE_SIGNATURE: 'profile.update-signature',
+  PROFILE_UPDATE_PHONE_NUMBER: 'profile.update-phone-number',
+  PROFILE_UPDATE_NAME: 'profile.update-name',
+  PROFILE_UPDATE_PROFILE_IMAGE: 'profile.update-profile-image',
```
[the remark this answers is not in the document]
Comment (tumbledwyer) [reply packages/commons/src/scopes.ts:96]: Thanks, added

File: packages/commons/src/scopes.ts [hunk 4]
```diff
@@ -8,103 +8,114 @@
...
-  'record.register',
+  RECORD_REGISTER: 'record.register',
 
-  // certify
-  'record.print-records',
-  'record.print-records-supporting-documents',
-  'record.export-records',
-  'record.print-issue-certified-copies',
-  'record.bulk-print-certified-copies',
+  //   // certify
```
[the remark this answers is not in the document]
Comment (tumbledwyer) [reply packages/commons/src/scopes.ts]: Thanks, fixed

File: packages/client/src/components/interface/Navigation.tsx [hunk 5]
```diff
@@ -672,7 +600,7 @@ const NavigationView = (props: IFullProps) => {
                   }
                 />
               )}
-              {hasPerformance && userDetails && (
+              {userDetails && hasAccess(WORKQUEUE_TABS.report) && (
```
[the remark this answers is not in the document]
Comment (tumbledwyer) [reply packages/client/src/components/interface/Navigation.tsx:603]: Yeah, even on develop there are 2 navigation items that point to performance. The 2nd one just has an id of report and a different icon but both call the `props.goToPerformanceViewAction`

File: packages/client/src/hooks/useNavigation.ts [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+  const routes = routeAccess.reduce((acc, group) => {
+    if (!hasAccess(group.scopes)) {
+      return acc
+    }
+    const groupAccess = {
+      name: group.name,
+      tabs: group.tabs
+        .filter((tab) => hasAccess(tab.scopes, tab.denyScopes))
+        .map((x) => ({
```
[the remark this answers is not in the document]
Comment (tumbledwyer) [reply packages/client/src/hooks/useNavigation.ts]: Done

File: packages/client/src/utils/gateway.ts [hunk 7]
```diff
@@ -89,6 +89,87 @@ export const scopes = [
 ] as const
```
[the remark this answers is not in the document]
Comment (Zangetsu101) [reply packages/client/src/utils/gateway.ts:89]: This is still being auto generated no? :sweat_smile:

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**